### ISSUE TITLE
(gnupg) Not approved

### ISSUE THREAD

**Issue by majkinetor** [chocolatey-community/chocolatey-coreteampackages#1333]
https://chocolatey.org/packages/gnupg

> gep13 (reviewer) on 26 Aug 2019 06:48:15 +00:00:
> Is this a duplicate of this package:
> * https://chocolatey.org/packages/gnupg-modern
>
> Or is that package going to be getting deprecated?

Its mentioned in Notes

>* Versions 2.1.16 to 2.2.12 of this tool are availalbe as gnupg-modern package (non embedded)

The PR on original repo is here:

* https://github.com/jtcmedia/chocolatey-packages/pull/32

This should get approved.

**Comment by gep13** [comment]
@majkinetor My question is still the same though...  Why create another package with a different id?  Should the other package id be deprecated, and point at the new one?

**Comment by ExE-Boss** [comment]
Probably, but I’m pretty sure that that has to be done by @wget.

**Comment by majkinetor** [comment]
@gep13, I answered everything already, not sure what doesn't compute ...

**Comment by gep13** [comment]
>@majkinetor said...
 I answered everything already, not sure what doesn't compute ...

The question regarding the existing package still hasn't been answered.  What is going to happen to it?  If this is a replacement package, that is fine, but I don't think the existing package should just be left.  If it is no longer going to be used, then it should be deprecated.

Also, if we are going to change the package id, then it should be updated everywhere, including the chocolateyInstall.ps1 file, which currently has this:

```
$packageName= 'gnupg-modern'
```

**Comment by ExE-Boss** [comment]
Actually, that’s `chocolateyUninstall.ps1`, which is probably unnecessary in this case, so I’m removing it in #1337.

**Comment by majkinetor** [comment]
> The question regarding the existing package still hasn't been answered. What is going to happen to it? If this is a replacement package, that is fine, but I don't think the existing package should just be left. If it is no longer going to be used, then it should be deprecated.

I already stated several times that fixing random repository isn't something we here do. We follwed the protocol for migration and there is nothing else we can do. If you have any further comments about this you should redirect them to appropriate place.

> Also, if we are going to change the package id, then it should be updated everywhere, including the chocolateyInstall.ps1 file, which currently has this

We are changing the package id becuase previous id was wrong (word `modern` is not something mentioned upstream) so discoverability and portability suffers. That is fairly obvious. Leftover was simply ommition that is already [fixed](https://github.com/chocolatey-community/chocolatey-coreteampackages/blob/master/automatic/gnupg/tools/chocolateyInstall.ps1#L6) several days ago here.

**Comment by ExE-Boss** [comment]
Now that #1337 has been merged, `gnupg@2.2.17` should probably be re‑published.

**Comment by majkinetor** [comment]
Did that

**Comment by gep13** [comment]
Until there is agreement on what is going to happen with the existing package (https://chocolatey.org/packages/gnupg-modern) i.e. who is taking responsibility for deprecating the package in favour of the new one, I am not going to be approving the new package, as there is no point in duplicating packages on the community repository.

**Comment by majkinetor** [comment]
Is that really how you want to resolve this? That is, blocking users to have the latest, the most secure and most stable package because of procedure nobody really cares about and has totally different goal then what end user is interested about,  making their environment use outdated and in this case insecure tools and thus dangerous for the benefit of internal jenitor operation? Especially because the things you requested could also be done after this one is approved at any time? Sounds to me a lot like failed priorities?

As this is common show stopper around here I sugest making deprecation procedure easier, such as maintainer clicking deprecate button on gallery. 

 /cc @ferventcoder

**Comment by gep13** [comment]
>@majkinetor said...
Especially because the things you requested could also be done after this one is approved at any time? Sounds to me a lot like failed priorities?

The problem with this approach is that if _not_ done now, the chances of them getting done afterwards diminish greatly, so yes, this is exactly how I would like to proceed.

>@majkinetor said...
As this is common show stopper around here I sugest making deprecation procedure easier, such as maintainer clicking deprecate button on gallery.

I think this is a great idea.  So that this doesn't get lost, I would suggest that this is added as a feature request on the [Gallery Repository](https://github.com/chocolatey/chocolatey.org)

**Comment by majkinetor** [comment]
> The problem with this approach is that if not done now, the chances of them getting done afterwards diminish greatly, so yes, this is exactly how I would like to proceed.

So lets be clear - the deprecation procedure sux which is why nobody wants to do it and yet you thunk that bevause of your stuborn insistence the whole world will suddenly have enlightment? Furthermore chocolatey employees break their own procedure all the time ( example: pauby has invokebuild package years after my package invoke-build). Whats the deal here? Some animals are more equal then others?

**Comment by majkinetor** [comment]
Anyway, I dont have energy for bike shedding here, fell free to block anything you want, I am out.

**Comment by gep13** [comment]
>@majkinetor said...
Furthermore chocolatey employees break their own procedure all the time ( example: pauby has invokebuild package years after my package invoke-build). Whats the deal here? Some animals are more equal then others?

There is no "deal".  Personally, I was not aware of this, and clearly this was missed during the initial review/approval of the package.  Where ever possible, I attempt to prevent duplicate of both packages, and effort, when it comes to packages.  As such, I have just raised this issue on @pauby repository:

https://github.com/pauby/ChocoPackages/issues/44

In my book, there is (and should be) no "favouritism" when it comes to packages that are pushed to chocolatey.org.  

Let's not forget that this issue started when the existing package was renamed from the original package id, causing a duplicate package.  If that hadn't been done, we wouldn't even be having this conversation.  Now, I don't disagree that renaming the package id is the correct course of action, all I am saying is that the work to tidy up the artifacts left from that decision need to be completed before moving forward.

**Comment by ExE-Boss** [comment]
I think we should approve this now and finish the deprecation in https://github.com/jtcmedia/chocolatey-packages/issues/36.

**Comment by majkinetor** [comment]
You mean, we should finish the deprecation in jtcmedia/chocolatey-packages#36 and approve this now ?🗡

**Comment by gep13** [comment]
@ExE-Boss I appreciate what you are trying to do in raising that request, but as you can see, it has gone unanswered for 12 days.  As a result, I am still hesitant about approving the new package until the old one is deprecated correctly.

**Comment by majkinetor** [comment]
Just out of curiosity, what happens if jtcmedia was hit by the bus (or simply doesn't care) ?

**Comment by majkinetor** [comment]
What if anybody doesn't care ?

**Comment by gep13** [comment]
Bottom line is, I, and other moderators, on chocolatey.org don't want to needlessly see duplication of packages pushed to chocolatey.org.

This is the exact situation that we have here.  If no one cares to tidy up the existing package, then simply put, I won't be approving the new package.  Another moderator may feel differently about this, but I would much rather see this situation fixed before approving the new package.  It has been my experience that if it isn't done now, it will never get done.

**Comment by majkinetor** [comment]
> If no one cares to tidy up the existing package, then simply put, I won't be approving the new package. 

Wow, A-mazing.

**Comment by jtcmedia** [comment]
I was asked to migrate this package to the coreteampackages some time ago and no longer have it in my own repository. How am I supposed to deprecate it in my repository when I no longer have it?

**Comment by ExE-Boss** [comment]
@jtcmedia You just create a stub‑package according to https://chocolatey.org/docs/how-to-deprecate-a-chocolatey-package, like what I’m doing in #1360.

**Comment by majkinetor** [comment]
https://chocolatey.org/packages/gnupg-modern/2.2.17

**Comment by gep13** [comment]
I have just approved the gnupg package.  @ExE-Boss thank you for taking the time to deprecate the old package.
